src/TestSuite/IntegrationTestTrait.php

```diff
@@ -512,6 +512,7 @@ protected function _sendRequest(array|string $url, string $method, array|string
             $response = $dispatcher->execute($request);
             $this->_requestSession = $request['session'];
             if ($this->_retainFlashMessages && $this->_flashMessages) {
+                $this->_requestSession->write('Flash', $this->_flashMessages);
```

Comment:
Does this fix the problem you were having in your tests?

Reply:
Yes, the error
```
1) App\Test\TestCase\Controller\ContactControllerTest::testIndexPostInvalid
TypeError: Cake\Utility\Hash::extract(): Argument #1 ($data) must be of type ArrayAccess|array, null given, 
called in /.../tests/TestCase/Controller/ContactControllerTest.php on line 57
```
goes away again